### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by mash2k3** [tomaae/homeassistant-portainer#23]
I have 5 environments
for this example lets name them a b c d e
The integration created 2 devices and named them Portioner Endpoints and Portainer a
It added all the container entities under Portainer a which is not a big deal for me.
But inside every container entity there is a attribute called environment that is set to e on all of them. 
That is the issue i wanted to point out because i have the same container running on more than 1 endpoint and i can to differentiate. 

## How to reproduce the issue
On initial install

## Expected behavior
Each container entity to have the correct environment name

## Screenshots
![image](https://github.com/tomaae/homeassistant-portainer/assets/1356023/456869c6-5aed-441f-8f52-9dda5a3e24e2)

## Software versions

 - Home Assistant version: 2023.10.5
 - Portainer integration version: v1.0.2
 - Portainer software version: 2.19.1

## Diagnostics data

I need time to go through the Diag data, at quick glance it had the portainer api key in plain text

**Comment by fanaticDavid** [comment]
I just discovered this really cool integration, but I can already confirm this bug. I have my Portainer-CE instance running on an Intel NUC, and I have 2 Portainer agents on my NAS and a Raspberry Pi. Every single container sensor has my NAS listed as its environment, which is incorrect. Sadly, #35 was auto-closed after going stale twice.

**Comment by PavelD** [comment]
I hope this issue should be fixed in the https://github.com/tomaae/homeassistant-portainer/pull/52

**Comment by tomaae** [comment]
No, that one is merged. Currently waiting for PR64 before I update integration for latest HA changes.

**Comment by bcutter** [comment]
Ah I see. Probably worth waiting, #64 looks interesting. But any estimation? It's been a while... 😃
